### PR TITLE
Testnet: implement read env options + image build + various refactoring

### LINKED ISSUE #668: Upgrade Dockerfile to Alpine 3.6
---
**By submitting this issue, I confirm the following:**

- I have read and understood the [contributor guide](https://github.com/anonimal/kovri-docs/blob/master/developer/CONTRIBUTING.md).
- I have checked that the issue I am reporting can be replicated or that the feature I am suggesting is not present.
- I have checked opened or recently closed [pull requests](https://github.com/monero-project/kovri/pulls) for existing solutions/implementations to my issue/suggestion.
---

Mainly making this as a reminder for me to do this.

### PR DESCRIPTION
**Pull request by anonimal**
Also bumps Alpine to 3.6 (resolves #668)

---
**By submitting this pull-request, I confirm the following:**

- I have read and understood the [Contributing Guide](https://github.com/anonimal/kovri-docs/blob/master/developer/CONTRIBUTING.md).
- I have checked that another pull-request for this purpose does not exist.
- I have considered and confirmed that this submission will be valuable to others.
- I accept that this submission may not be used and that this pull-request may be closed by the will of the maintainer.
- I give this submission freely under the BSD 3-clause license.
---

### CODE REVIEW

File: contrib/docker/testnet.sh [hunk 1]
```diff
@@ -1,12 +1,7 @@
 #!/bin/bash
 
-# Get/set environment
-KOVRI_DIR=${KOVRI_DIR:-"/tmp/kovri"}
-KOVRI_WORKSPACE=${KOVRI_WORKSPACE:-"${KOVRI_DIR}/build/testnet"}
```
Comment (MoroccanMalinois) [inline contrib/docker/testnet.sh]: This variable is redefined in `Create` but it's also used in `Destroy`

File: contrib/docker/testnet.sh [hunk 2]
```diff
@@ -33,15 +28,73 @@ fi
...
+  catch "Could not access $KOVRI_REPO"
+  # Set tag
+  hash git 2>/dev/null
+  if [[ $? -ne 0 ]]; then
+    echo "git is not installed, using default tag"
+    _docker_tag=":latest"
+  else
+    _docker_tag=":$(git rev-parse --short HEAD)"
+  fi
+  KOVRI_IMAGE=${KOVRI_IMAGE:-"geti2p/kovri"}${_docker_tag}
```
Comment (anonimal) [inline contrib/docker/testnet.sh]: Oops (that should've been inside the brace)! Thanks @MoroccanMalinois

File: contrib/docker/testnet.sh (hunk 1 above)
Comment (anonimal) [inline contrib/docker/testnet.sh]: Fixed in 358bf8f, this PR number noted in git log. Thanks @MoroccanMalinois.

**Comment by anonimal** [comment]
Three more commits pushed since most recent review.

**Pull request merged**